beartype/_protocol.py

```diff
@@ -0,0 +1,236 @@
+# --------------------( LICENSE                           )--------------------
+# Copyright (c) 2014-2022 Beartype authors.
+# See "LICENSE" for further details.
```

Comment:
I know you have particular stylistic preferences for organization/naming/commenting. If you're okay with it, I won't attempt to meet all those preferences here. Rather, I'll focus on getting the rough structure in place first.

Reply:
Putting the comfy chair back for now. Happy to ~revisit~ _continue to avoid_ later as necessary. 😜